src/python_testing/TC_SUTestBase.py

```diff
@@ -0,0 +1,480 @@
+            providerNodeID=provider_node_id,
+            endpoint=endpoint,
+            fabricIndex=controller.fabricId
+        )
+
+        # Create the OTA Provider Attribute
+        ota_providers_attr = Clusters.OtaSoftwareUpdateRequestor.Attributes.DefaultOTAProviders(value=[provider_location_struct])
+
+        # Write the Attribute
+        resp = await controller.WriteAttribute(
+            attributes=[(endpoint, ota_providers_attr)],
+            nodeid=requestor_node_id,
+        )
+        asserts.assert_equal(resp[0].Status, Status.Success, "Failed to write Default OTA Providers Attribute")
+
+        # Read Updated OTAProviders
+        after_otap_info = await self.read_single_attribute_check_success(
+            dev_ctrl=controller,
+            cluster=Clusters.OtaSoftwareUpdateRequestor,
+            attribute=Clusters.OtaSoftwareUpdateRequestor.Attributes.DefaultOTAProviders
+        )
+        logger.info(f"OTA Providers List: {after_otap_info}")
+
+    async def verify_version_applied_basic_information(self, controller: ChipDeviceCtrl, node_id: int, target_version: int):
+        """Verify the version from the BasicInformationCluster and compares against the provider target version.
+
+        Args:
+            controller (ChipDeviceCtrl): Controller 
+            node_id (int): Node to request
+            target_version (int): Version to compare
+        """
+
+        basicinfo_softwareversion = await self.read_single_attribute_check_success(
+            dev_ctrl=controller,
+            cluster=Clusters.BasicInformation,
+            attribute=Clusters.BasicInformation.Attributes.SoftwareVersion,
+            node_id=node_id)
+        asserts.assert_equal(basicinfo_softwareversion, target_version,
+                             f"Version from basic info cluster is not {target_version}, current cluster version is {basicinfo_softwareversion}")
+
+    def get_downloaded_ota_image_info(self, ota_path='/tmp/test.bin') -> dict:
+        """Return the data of the downloaded image from the provider.
+
+        Args:
+            ota_path (str, optional): _description_. Defaults to '/tmp/test.bin'.
+
+        Returns:
+            dict: Dict with the image info.
+        """
+        ota_image_info = {
+            "path": ota_path,
+            "exists": False,
+            "size": 0,
+        }
+        try:
+            ota_image_info['size'] = path.getsize(ota_path)
+            ota_image_info['exists'] = True
+        except OSError:
+            logger.info(f"OTA IMAGE at {ota_path} does not exists")
+            return ota_image_info
+
+        return ota_image_info
+
+    def verify_state_transition_event(self,
+                                      event_report: Clusters.OtaSoftwareUpdateRequestor.Events.StateTransition,
+                                      previous_state,
+                                      new_state,
```

Comment:
nit: I would rename this to the `expected_previous_state` and `expected_new_state` correspondingly